### PR TITLE
12037 phalanx location

### PR DESCRIPTION
**Pull request by cip123**
## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/12037

## Tests Performed:
- It compiles and I tested this change on MacOs

## How to Test the Changes:
1. As a GameMaster type `.go c 46615`
2. Buy  6 dark iron ale mugs which from Plugger Spazzring (the gnome with a wizard hat) angry.
3. Give the 6 mugs to Rocknot. He will become drunk, break the kegs and that will upset Phalanx.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by tkn963** [comment]
There shouldn't be `acore_world.` in the sql queries as not everyone uses that name for their world database.

File: data/sql/updates/pending_db_world/phalanx-location.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+UPDATE creature SET position_x = 848.066, position_y = -230.809, position_z = -43.698, orientation = 1.18512 WHERE guid = 46615;
+
+INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action, action_chance, wpguid) VALUES (32365006, 1, 862.813, -220.542, -43.7036, 1, 0, 2, 0, 100, 0);
+INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action, action_chance, wpguid) VALUES (32365006, 2, 869.566, -226.553, -43.6977, -0.654868, 0, 2, 0, 100, 0);
+INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action, action_chance, wpguid) VALUES (32365006, 3, 867.15, -222.741, -43.6977, 2.18024, 0, 2, 0, 100, 0);
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/phalanx-location.sql:6]:
```
UPDATE creature SET position_x = 848.066, position_y = -230.809, position_z = -43.698, orientation = 1.18512 WHERE guid = 46615;

DELETE FROM `waypoint_data` WHERE `id` = 32365006 AND `point` IN (1, 2, 3);
INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action, action_chance, wpguid) VALUES 
(32365006, 1, 862.813, -220.542, -43.7036, 1, 0, 2, 0, 100, 0),
(32365006, 2, 869.566, -226.553, -43.6977, -0.654868, 0, 2, 0, 100, 0),
(32365006, 3, 867.15, -222.741, -43.6977, 2.18024, 0, 2, 0, 100, 0);
```

**Review by acidmanifesto — CHANGES_REQUESTED** [review]
i will get with the staff about the id numbering.

File: data/sql/updates/pending_db_world/phalanx-location.sql (hunk 1 above)
Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/phalanx-location.sql:6]: thats a reformat of the sql to mean [Azerothcore's SQL Standards ](https://www.azerothcore.org/wiki/sql-standards).

File: data/sql/updates/pending_db_world/phalanx-location.sql [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+  AND `point` IN (1, 2, 3);
 
-INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action, action_chance, wpguid) VALUES (32365006, 1, 862.813, -220.542, -43.7036, 1, 0, 2, 0, 100, 0);
-INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action, action_chance, wpguid) VALUES (32365006, 2, 869.566, -226.553, -43.6977, -0.654868, 0, 2, 0, 100, 0);
-INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action, action_chance, wpguid) VALUES (32365006, 3, 867.15, -222.741, -43.6977, 2.18024, 0, 2, 0, 100, 0);
+INSERT INTO waypoint_data (id, point, position_x, position_y, position_z, orientation, delay, move_type, action,
+                           action_chance, wpguid)
+VALUES (32365006, 1, 862.813, -220.542, -43.7036, 1, 0, 2, 0, 100, 0),
+       (32365006, 2, 869.566, -226.553, -43.6977, -0.654868, 0, 2, 0, 100, 0),
+       (32365006, 3, 867.15, -222.741, -43.6977, 2.18024, 0, 2, 0, 100, 0);
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/phalanx-location.sql:17]: we need a line break at the end of this.
tldr; hit enter at the end and save.

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by acidmanifesto** [comment]
@azerothcore/testers

**Review by acidmanifesto — APPROVED** [review]

**Review by Nyeriah — CHANGES_REQUESTED** [review]
Thanks for the PR, however, the SQL still needs to be updated to comply with the SQL submission format. I've updated it for you here: https://github.com/azerothcore/azerothcore-wotlk/pull/12127#discussion_r909592954

**Comment by cip123** [comment]
> @cip123 how were the positions gps obtained in this pr?

I used GM mode to position myself in BRD and got the GPS locations.

**Comment by Maelthyrr** [comment]
Tested and work. Maybe one suggestion, it's seems that phalanx yell something during the path, could you see and add this ? It's to avoid another PR just for one emote...

**Comment by acidmanifesto** [comment]
> > @cip123 how were the positions gps obtained in this pr?
> 
> I used GM mode to position myself in BRD and got the GPS locations.

so you only reused the exsisting gps already?

**Comment by cip123** [comment]
No, I approximated new GPS locations by positioning myself in the corner
and then at the door.

**Comment by Gultask** [comment]
Spawn location: 
```
[3] Position: X: 847.84796 Y: -230.06673 Z: -43.61398
[3] Orientation: 1.640609502792358398
```

Path:
```
-- Pathing for Phalanx Entry: 9502
SET @NPC := XXXXXX;
SET @PATH := @NPC * 10;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`visibilityDistanceType`,`auras`) VALUES (@NPC,@PATH,0,0,1,0,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,865.3834,-219.31128,-43.7037,0,0,0,0,100,0),
(@PATH,2,867.29456,-222.6452,-43.72592,0,0,0,0,100,0),
(@PATH,3,867.29456,-222.6452,-43.72592,2.007128715515136718,0,0,0,100,0);
```

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp [hunk 3]
```diff
@@ -435,6 +435,11 @@ enum PhalanxSpells
     SPELL_MIGHTYBLOW                    = 14099
 };
 
+enum PhalanxPaths
+{
+    PHALANX_PATH_ID                     = 32365006,
```
Comment (Winfidonarleyan) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp:440]: constexpr

**Comment by Maelthyrr** [comment]
@cip123 is it abandon ?

**Comment by cip123** [comment]
No, I'll take care of it.

**Review by acidmanifesto — CHANGES_REQUESTED** [review]
https://github.com/azerothcore/azerothcore-wotlk/pull/12127/files#r928296761
Please address the Review

**Comment by cip123** [comment]
@acidmanifesto  Please see my previous comment: https://github.com/azerothcore/azerothcore-wotlk/pull/12127#issuecomment-1220869655

**Comment by acidmanifesto** [comment]
> @acidmanifesto Please see my previous comment: [#12127 (comment)](https://github.com/azerothcore/azerothcore-wotlk/pull/12127#issuecomment-1220869655)

i am not entirely certain what this has to do with me.

**Comment by acidmanifesto** [comment]
Marking as WIP, please address the reviews and update the PR.
